plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplit.java

```diff
@@ -35,6 +35,7 @@ public record BigQuerySplit(
         List<BigQueryColumnHandle> columns,
         long emptyRowsToGenerate,
         Optional<String> filter,
+        Optional<Long> limit,
```

Comment:
Any specific reason on why `BigQuerySplit` needs to maintain this limit ? `BigQueryTableHandle` already has this information - and it is available when creating the PageSource - Can we use them from the table handle ?